src/librustc/mir/repr.rs

```diff
@@ -699,6 +733,9 @@ pub enum StatementKind<'tcx> {
 
     /// End the current live range for the storage of the local.
     StorageDead(Lvalue<'tcx>),
+
+    /// No-op. Useful for deleting instructions without affecting statement indices.
+    Nop,
```

Comment:
It's in [`src/librustc_borrowck/borrowck/mir/patch.rs`](https://github.com/rust-lang/rust/blob/master/src/librustc_borrowck/borrowck/mir/patch.rs) (because drop elaboration in there. IMO `rustc_borrowck` should be replaced with a MIR version that doesn't live in a `rustc_borrowck` crate but that's besides the point).

It doesn't have the ability to remove statements because it wasn't needed.
I'm not entirely clear on whether that functionality could be easily added. cc @arielb1